From 21a63831b0cbc8834fda2a9c3afc99acbb3071b3 Mon Sep 17 00:00:00 2001
From: Bryan Burke <bburke@baburke.net>
Date: Sat, 3 Aug 2019 19:48:58 -0700
Subject: [PATCH 4/4] enhancing solarized colors

---
 autoload/airline/themes/solarized.vim | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/autoload/airline/themes/solarized.vim b/autoload/airline/themes/solarized.vim
index 702eda9..57debfd 100644
--- a/autoload/airline/themes/solarized.vim
+++ b/autoload/airline/themes/solarized.vim
@@ -18,7 +18,7 @@ function! airline#themes#solarized#refresh()
   " Extended base16 support by @cuviper.
   " Via https://github.com/blueyed/vim-colors-solarized/commit/92f2f994 /
   " https://github.com/cuviper/vim-colors-solarized.
-  if s:ansi_colors && get(g:, 'solarized_base16', 0)
+  if has("termguicolors") || (s:ansi_colors && get(g:, 'solarized_base16', 0))
     let s:base03  = {'t': 0,  'g': "#002b36"}  " Base 00
     let s:base02  = {'t': 18, 'g': "#073642"}  " Base 01
     let s:base01  = {'t': 19, 'g': "#586e75"}  " Base 02
@@ -27,14 +27,14 @@ function! airline#themes#solarized#refresh()
     let s:base1   = {'t': 7,  'g': "#93a1a1"}  " Base 05
     let s:base2   = {'t': 21, 'g': "#eee8d5"}  " Base 06
     let s:base3   = {'t': 15, 'g': "#fdf6e3"}  " Base 07
-    let s:yellow  = {'t': 3,  'g': "#dc322f"}  " Base 0A
+    let s:yellow  = {'t': 3,  'g': "#b58900"}  " Base 0A
     let s:orange  = {'t': 16, 'g': "#cb4b16"}  " Base 09
-    let s:red     = {'t': 1,  'g': "#b58900"}  " Base 08
-    let s:magenta = {'t': 17, 'g': "#859900"}  " Base 0F
-    let s:violet  = {'t': 5,  'g': "#2aa198"}  " Base 0E
+    let s:red     = {'t': 1,  'g': "#dc322f"}  " Base 08
+    let s:magenta = {'t': 17, 'g': "#d33682"}  " Base 0F
+    let s:violet  = {'t': 5,  'g': "#6c71c4"}  " Base 0E
     let s:blue    = {'t': 4,  'g': "#268bd2"}  " Base 0D
-    let s:cyan    = {'t': 6,  'g': "#6c71c4"}  " Base 0C
-    let s:green   = {'t': 2,  'g': "#d33682"}  " Base 0B
+    let s:cyan    = {'t': 6,  'g': "#2aa198"}  " Base 0C
+    let s:green   = {'t': 2,  'g': "#859900"}  " Base 0B
   else
     let s:base03  = {'t': s:ansi_colors ?   8 : (s:tty ? '0' : 234), 'g': '#002b36'}
     let s:base02  = {'t': s:ansi_colors ? '0' : (s:tty ? '0' : 235), 'g': '#073642'}
-- 
2.22.0

